cmake/all.cmake

```diff
@@ -162,6 +166,11 @@ set(CATKIN_GLOBAL_LIBEXEC_DESTINATION lib)
 set(CATKIN_GLOBAL_PYTHON_DESTINATION ${PYTHON_INSTALL_DIR})
 set(CATKIN_GLOBAL_SHARE_DESTINATION share)
 
+# This was removed, but is required for Windows
```

Comment:
Currently `configure_shared_library_build_settings()` is being called only conditionally on Windows. That seems wrong to me. Shouldn't this be done for all platforms?

Anyway this change has the potential of breaking existing use cases.